### PR TITLE
Fix `strict` flag mapping

### PR DESCRIPTION
**Pull request by pavansokkenagaraj**
fix `strict` flag mapping
- if `strict` is valid BoolString value, map accordingly
- else, map false as default and ignore parse error

### CODE REVIEW

File: pkg/multitype/boolstring_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		}, {
+			name:    "expect false when StrVal is 'true'",
+			fields:  fields{Type: String, StrVal: "true"},
+			want:    true,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'true'",
+			fields:  fields{Type: String, StrVal: "true"},
+			want:    true,
+			wantErr: false,
```
Comment (garcialuis) [inline pkg/multitype/boolstring_test.go]: These are duplicated, one should be "expect true when StrVal is 'true'"?

File: pkg/multitype/boolstring.go [hunk 2]
```diff
@@ -116,3 +117,21 @@ func (boolstr *BoolOrString) Fuzz(c fuzz.Continue) {
 		c.Fuzz(&boolstr.StrVal)
 	}
 }
+
+// BoolOrDefaultFalse returns bool val, if strValu is parsed returns parsed value  else false as default when parse error
+func (boolstr *BoolOrString) BoolOrDefaultFalse() bool {
+	val, _ := boolstr.Bool()
+	return val
+}
```
Comment (laverya) [inline pkg/multitype/boolstring.go]:
```suggestion
func (boolstr *BoolOrString) BoolOrDefaultFalse() bool {
	val, err := boolstr.Bool()
	if err != nil {
		return false
	}
	return val
}
```
you shouldn't rely on the behavior of something returning non-nil error

**Review by garcialuis — CHANGES_REQUESTED** [review]

File: pkg/multitype/boolstring_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		}, {
+			name:    "expect false when BoolVal is false",
+			fields:  fields{Type: Bool, BoolVal: false},
+			want:    false,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'true'",
+			fields:  fields{Type: String, StrVal: "true"},
+			want:    true,
+			wantErr: false,
```
Comment (laverya) [inline pkg/multitype/boolstring_test.go]:
```suggestion
			name:    "expect false when StrVal is 'false'",
			fields:  fields{Type: String, StrVal: "false"},
			want:    false,
			wantErr: false,
```

File: pkg/multitype/boolstring_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			if (err != nil) != tt.wantErr {
+				t.Errorf("BoolOrString.Bool() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("BoolOrString.Bool() = %v, want %v", got, tt.want)
+			}
+			if got := boolstr.BoolOrDefaultFalse(); got != tt.want {
+				t.Errorf("BoolOrString.BoolOrDefaultFalse() = %v, want %v", got, tt.want)
+			}
```
Comment (laverya) [inline pkg/multitype/boolstring_test.go]:
```suggestion
			req := require.New(t)
			boolstr := &BoolOrString{
				Type:    tt.fields.Type,
				BoolVal: tt.fields.BoolVal,
				StrVal:  tt.fields.StrVal,
			}
			got, err := boolstr.Bool()
			if tt.wantErr {
				req.Error(err)
				return
			}
			req.NoError(err)
			req.Equal(tt.want, got)
			req.Equal(tt.want, boolstr.BoolOrDefaultFalse())
```

Comment (laverya) [reply pkg/multitype/boolstring_test.go]: this uses testify/require, which you'd need to import, but we use elsewhere in the project

File: pkg/multitype/boolstring_test.go (hunk 1 above)
Comment (pavansokkenagaraj) [reply pkg/multitype/boolstring_test.go]: nice catch, thanks

File: pkg/multitype/boolstring_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			fields:  fields{Type: Bool, BoolVal: false},
+			want:    false,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'false'",
+			fields:  fields{Type: String, StrVal: "false"},
+			want:    false,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'true'",
```
Comment (garcialuis) [inline pkg/multitype/boolstring_test.go]:
```suggestion
			name:    "expect true when StrVal is 'true'",
```

**Review by garcialuis — APPROVED** [review]

**Pull request merged**